src/main/java/com/google/sps/data/BusinessesService.java

```diff
@@ -111,4 +128,109 @@ private void addListingToBusinesses(PlacesSearchResult place) {
     allBusinesses.add(new Listing(name, formattedAddress, 
           placeLocation, rating, photos, types));
   }
-}
\ No newline at end of file
+
+  public void checkNumberOfLocationsOfBusiness() {
+    GeoApiContext context = new GeoApiContext.Builder()
+            .apiKey(KEY)
+            .build();
+    businesses =  allBusinesses.iterator(); 
+    try {
+      while (businesses.hasNext()) {
+        currentBusiness = businesses.next();
+        TextSearchRequest request = new TextSearchRequest(context)
+                                            .query(currentBusiness.getName())
+                                            .location(latLng)
+                                            .radius(50000);
+        PlacesSearchResult[] similarBusinessesInTheArea = request
+                                                            .await()
+                                                            .results;
+        if (similarBusinessesInTheArea.length > 1){
+          checkBusinessThroughLinkedin(currentBusiness.getName(), similarBusinessesInTheArea);
+        }
+      }
+    } catch(GeneralSecurityException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(IOException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(ApiException e) {
+      LOGGER.warning(e.getMessage());
+    } catch(InterruptedException e) {
+      LOGGER.warning(e.getMessage());
+    } 
+  }
+  
+  private void checkBusinessThroughLinkedin(String currentBusinessName, 
+                          PlacesSearchResult[] similarBusinessesInTheArea) 
+                              throws GeneralSecurityException, IOException {
+    String cx = "REDACTED"; 
+    Customsearch cs = new Customsearch.Builder(
+                                GoogleNetHttpTransport.newTrustedTransport(), 
+                                JacksonFactory.getDefaultInstance(), 
+                                null) 
+                                .setApplicationName("linkedinSearch") 
+                                .setGoogleClientRequestInitializer(
+                                      new CustomsearchRequestInitializer(KEY)) 
+                                .build();
+
+    Customsearch.Cse.List list = cs.cse().list(currentBusinessName).setCx(cx); 
+    List<Result> searchJsonResults = list.execute().getItems();
+    String[] numberOfFollowers;
+    int companyFollowers = 0;
+    if (searchJsonResults!=null && searchJsonResults.size() != 0) {
+      Result linkedinBusiness = searchJsonResults.get(0);
+      String businessDescription = (String) linkedinBusiness.getPagemap()
+                                                            .get("metatags")
+                                                            .get(0)
+                                                            .get("og:description");
+      if(businessDescription.indexOf(startIndex) != -1 
```

Comment:
nit: A comment here may be helpful to explain what the -1 is doing, etc.

Reply:
Not sure if I should as, indexOf returns -1 instead of null if something is not found in the string. I think this is something someone can easily figure out or it is already known